pkg/middleend/middleend.go

```diff
@@ -295,30 +290,24 @@ func (s *Server) EncryptedVolumeStats(_ context.Context, in *pb.EncryptedVolumeS
 	}}, nil
 }
 
+func (s *Server) verifyEncryptedVolume(volume *pb.EncryptedVolume) error {
+	switch {
+	case volume.Cipher == pb.EncryptionType_ENCRYPTION_TYPE_AES_XTS_256:
+		return nil
+	case volume.Cipher == pb.EncryptionType_ENCRYPTION_TYPE_AES_XTS_128:
+		return nil
+	default:
+		return fmt.Errorf("only AES_XTS_256 and AES_XTS_128 are supported")
+	}
+}
+
 func (s *Server) getAccelCryptoKeyCreateParams(volume *pb.EncryptedVolume) models.AccelCryptoKeyCreateParams {
 	var params models.AccelCryptoKeyCreateParams
 
-	switch volume.Cipher {
-	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_128:
-		fallthrough
-	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_192:
-		fallthrough
-	case pb.EncryptionType_ENCRYPTION_TYPE_AES_CBC_256:
-		params.Cipher = "AES_CBC"
-		params.Key = hex.EncodeToString(volume.Key)
```

Comment:
@artek-koltun I understand DPSK only supports `AES_XTS`
But what if some vendors supports also `AES_CBC` ?
Obviously they can re-implement this entire service, but it would be nice, for example, if they could just override very small part...

WDYT ?

Reply:
yes, I thought about it, but decided to postpone it until we really need that functionality for other DPUs.
However, if you want to see that functionality right now, I can add it. No issues